src/types/IProtocolAdapter.ts

```diff
@@ -35,6 +36,8 @@ export interface IProtocolAdapter {
    */
   productId: string
 
+  adaptersController: AdaptersController
```

Comment:
As discussed, I think it would be beneficial to stop adding things here at public level and add instead a single toolbox class that exposes what we need:
- Methods from adapter controller
- ChainId
- Provider
- Anything new we want to add

If we do that, then we won't have to change every adapter that implements this interface directly every time we add a utility to that class (unless that adapter needs to use it).